bioio_base/reader.py

```diff
@@ -1070,6 +1083,89 @@ def get_xarray_dask_stack(self, **kwargs: Any) -> xr.DataArray:
         """
         return transforms.generate_stack(self, mode="xarray_dask_data", **kwargs)
 
+    # --- New embedded metadata properties for the manifest ---
+    def _get_dim_size(self, dim: str) -> Optional[int]:
+        if dim in self.dims.order:
+            idx = self.dims.order.index(dim)
+            return self.shape[idx]
+        return None
+
+    @property
+    @embedded_metadata(MetadataLabels.DIMENSIONS_PRESENT.value)
+    def dimensions_present(self) -> Optional[str]:
+        return self.dims.order
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_C.value)
+    def image_size_c(self) -> Optional[int]:
+        return self._get_dim_size("C")
```

Comment:
Instead of "C" and the other dimensions, there is a constant for them you can use

Reply:
is it getting a little bit weird that we keep adding more and more public api methods to return the same stuff?  I wonder why we don't just have one single public method to return the whole batch of metadata that we want to define?